File: problems/1-easy/226-Invert-Binary-Tree/solution.cpp
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};


TreeNode* invertTree(TreeNode* root) {
    if (root == nullptr) {
        return root;
    }
    
    dfs(root);
    return root;
}

void dfs(TreeNode* node) {
    if (node->left) {
        dfs(node->left);
    }
    if (node->right) {
        dfs(node->right);
    }

    TreeNode* temp = node->left;
    node->left = node->right;
    node->right = temp;
}

File: problems/1-easy/226-Invert-Binary-Tree/solution.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root:
            self.dfs(root)
        return root

    def dfs(self, current):
        if current.left:
            self.dfs(current.left)

        if current.right:
            self.dfs(current.right)

        current.left, current.right = current.right, current.left
